dandi/models.py

```diff
@@ -113,6 +113,185 @@ def diff_models(model1, model2):
             print(f"{field} is different")
 
 
+DATACITE_CONTRTYPE = [
+    "ContactPerson",
+    "DataCollector",
+    "DataCurator",
+    "DataManager",
+    "Distributor",
+    "Editor",
+    "HostingInstitution",
+    "Producer",
+    "ProjectLeader",
+    "ProjectManager",
+    "ProjectMember",
+    "RegistrationAgency",
+    "RegistrationAuthority",
+    "RelatedPerson",
+    "Researcher",
+    "ResearchGroup",
+    "RightsHolder",
+    "Sponsor",
+    "Supervisor",
+    "WorkPackageLeader",
+    "Other",
+]
+
+
+DATACITE_IDENTYPE = [
+    "ARK",
+    "arXiv",
+    "bibcode",
+    "DOI",
+    "EAN13",
+    "EISSN",
+    "Handle",
+    "IGSN",
+    "ISBN",
+    "ISSN",
+    "ISTC",
+    "LISSN",
+    "LSID",
+    "PMID",
+    "PURL",
+    "UPC",
+    "URL",
+    "URN",
+    "w3id",
+]
+DATACITE_MAP = dict([(el.lower(), el) for el in DATACITE_IDENTYPE])
+
+
+def to_datacite(meta):
+    dandiset_id = meta.identifier.split(":")[1]
+    doi = meta.doi
+
+    attributes = {}
+    attributes["identifiers"] = [
+        # TODO: the first elementis ignored, not sure how to fix it...
+        {"identifier": f"https://doi.org/{doi}", "identifierType": "DOI"},
+        {
+            "identifier": f"https://identifiers.org/DANDI:{dandiset_id}/{meta.version}",
+            "identifierType": "URL",
+        },
+    ]
+
+    attributes["doi"] = doi
+    attributes["titles"] = [{"title": meta.name}]
+    attributes["descriptions"] = [
+        {"description": meta.description, "descriptionType": "Abstract"}
+    ]
+    attributes["publisher"] = "DANDI Archive"
+    attributes["publicationYear"] = str(meta.datePublished)
```

Comment:
Why is this called `publicationYear` when it's being set to a timestamp?  Is that intentional?

Reply:
moved to `meta.datePublished.year`